pkgs/development/tools/detekt/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetchFromGitHub, unzip, makeWrapper, jre_headless }:
+
+stdenv.mkDerivation rec {
+  pname = "detekt";
+  version = "1.22.0";
+
+  src = fetchFromGitHub {
+    owner = "detekt";
+    repo = "detekt";
+    rev = "v${version}";
+    sha256 = "08jbnn3q3c7vp6jmj3gz67d5v9jlgd9f3dr706cbk8q2v9719gqh";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  preBuild = ''
+    export JAVA_HOME=${jre_headless}
+  '';
+
+  installPhase = ''
+    ./gradlew :detekt-cli:distZip
```

Comment:
./gradlew :detekt-cli:distZip is downloading from the internet. This won't work because the sandbox is now enabled by default on all operating systems which means this will always fail. It's working for you because at Motive, we have the sandbox disabled at the moment.

Reply:
Thanks for the info and all the help. I looked up similar projects inside nixpkgs and figured that I have to download the dependencies before the install phase and then fetch the deps from a local repository in the installphase. 
I will give this another try when I have more time. So closing this PR for now.